adapters/backend/v1/kafka_integration_test.go

```
@@ -25,20 +29,61 @@ const kafkaTestMaxMessageBytes = 128 * 1024 * 1024 // 128 MB, headroom above the
+			nat.Port("9092/tcp"): []nat.PortBinding{{HostIP: "127.0.0.1", HostPort: strconv.Itoa(hostPort)}},
+		}
 	})
+}
 
-	broker, err := container.KafkaSeedBroker(ctx)
+// freeKafkaHostPort asks the kernel for an unused port and releases it, so the broker can be
+// rebuilt on the same address. Racy in principle, but against the whole machine rather than
+// against a hardcoded range that overlaps the pulsar tests'.
+func freeKafkaHostPort(t *testing.T) int {
+	t.Helper()
+	listener, err := net.Listen("tcp", "127.0.0.1:0")
```

golangci-lint (2.12.2)

[error] net.Listen must not be called. use (*net.ListenConfig).Listen

(noctx)